spinnman/processes/read_iobuf_process.py

```diff
@@ -37,22 +37,48 @@ def __init__(self, connection_selector):
         # read = list of (x, y, p, n, next_address, first_read_size)
         self._next_reads = list()
 
-    def handle_iobuf_address_response(self, iobuf_size, x, y, p, response):
+    def _request_iobuf_address(self, iobuf_size, processor_address):
```

Comment:
The code in here does very much passing around of the “processor address” (an `x`,`y`,`p` triple) without interpreting it. My thought was that we should stop unpacking and repacking this info as it bounces around all these methods that are internal to the `ReadIOBufProcess` class. While in general a processor address is a concept that's used a lot elsewhere (generating much memory churn), these are all methods that are effectively internal to SpiNNMan. (I couldn't find any code outside of SpiNNMan and its tests that used the processes directly; the public transceiver API does not.)

I don't like packing a tuple to use it as a key that is then picked up by iteration and unpacked and sent into a method that uses some bits and repacks as a tuple for some other reason and … Well, long experience tells me that churning memory with lots of small objects is surprisingly expensive, especially when we don't actually need to. Why don't we just admit that there really is a (relative-to-a-machine) processor address concept and that we actually use it all over? (There's also a chip address concept lurking in there; I don't think I've made any use of it in my changes, but it really does come up a lot. Ethernet chip addresses would be a subclass of it.)

Reply:
So this is a reason for a named tuple to exist - although these are also a bit nasty in their documentation.  They at least enforce the parameters of the tuple and somewhat document what the tuple contains.  Note that the processor_address is passed in from the transceiver in some cases; it might be more acceptable if it were *only* internally used.